convert-sg-db.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python
+
+import json
+import commonmark
+
+
+def create_biblio(db):
+    biblio = {}
+
+    for workstream in db["workstreams"]:
+        for standard in workstream["standards"]:
+            reference, contents = standard_or_idea_to_biblio(standard)
+            contents["title"] += " Standard"
+            biblio[reference] = contents
+
+    for idea in db["ideas"]:
+        reference, contents = standard_or_idea_to_biblio(idea)
+        contents["status"] = "Living Idea"
+        biblio[reference] = contents
+
+    for biblio_item in db["biblio"]:
+        reference, contents = biblio_item_to_biblio(biblio_item)
+        biblio[reference] = contents
+
+    string = json.dumps(biblio, ensure_ascii=False, allow_nan=False, sort_keys=True, indent=2)
+    string += "\n"
+    open("resources.whatwg.org/biblio.json", "w").write(string)
+
+def standard_or_idea_to_biblio(document):
+    return (document["reference"], {
+        "authors": [author_to_biblio(author) for author in document["authors"]],
+        "href": document["href"],
+        "title": document["name"]
+    })
+
+def author_to_biblio(person):
+    return person["name"]
+
+def biblio_item_to_biblio(item):
+    contents = {
+        "href": item["href"],
+        "title": item["title"]
+    }
+    if len(item["authors"]):
+        contents["authors"] = [author_to_biblio(author) for author in item["authors"]]
+    if "obsoletedBy" in item:
+        contents["obsoletedBy"] = item["obsoletedBy"]
+    return (item["reference"], contents)
+
+
+def create_spec_whatwg_org(db, template, copyright):
+    content = "<p>The WHATWG works on a number of technologies that are fundamental parts of the web platform. They are organised somewhat arbitrarily based on the preferences of those editing the standard for those technologies.</p>\n\n"
+    content += "<dl>"
+    for workstream in db["workstreams"]:
+        for standard in workstream["standards"]:
+            content += standard_or_idea_to_html(standard)
+    content += "</dl>\n\n<p>The WHATWG also works on a <a href=\"https://idea.whatwg.org/\">number of ideas</a> that aspire to become standards one day.</p>\n\n"
+    content += copyright
+
+    output = template.replace("@TITLE_GOES_HERE@", "Standards")
+    output = output.replace("@CONTENT_GOES_HERE@", content)
+    output = output.replace("<a href=\"https://spec.whatwg.org/\">Standards</a>", "<a>Standards</a>")
+
+    open("spec.whatwg.org/index.html", "w").write(output)
+
+def standard_or_idea_to_html(document):
+    output = "\n <dt><a href=\"{0}\">{1}</a></dt>".format(document["href"], document["name"])
+    return output + "\n <dd>{}</dd>\n".format(commonmark.commonmark(document["description"])[:-1])
```

Comment:
Nit: this might be nicer:

```python
# TODO add a comment for the [:-1] here
description = commonmark.commonmark(document["description"])[:-1]

return "\n <dt><a href=\"{0}\">{1}</a></dt>\n <dd>{2}</dd>\n"
       .format(document["href"], document["name"], description)
```

Either way, the `[:-1]` is confusing and a comment explaining it would probably help.